client/src/connection/itc/CodeRunner.ts

```diff
@@ -0,0 +1,92 @@
+// Copyright © 2024, SAS Institute Inc., Cary, NC, USA.  All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import { commands } from "vscode";
+
+import { ITCSession } from ".";
+import { LogLine, getSession } from "..";
+import { useRunStore } from "../../store";
+
+class CodeRunner {
+  protected executionInterval: ReturnType<typeof setInterval>;
+
+  public async runCode(
+    code: string,
+    startTag: string = "",
+    endTag: string = "",
+  ): Promise<string> {
+    // If we're already executing code, lets wait for it
+    // to finish up.
+    if (useRunStore.getState().isExecutingCode) {
+      await new Promise((resolve) => {
+        if (this.executionInterval) {
+          clearInterval(this.executionInterval);
+        }
+
+        this.executionInterval = setInterval(() => {
+          if (!useRunStore.getState().isExecutingCode) {
+            clearInterval(this.executionInterval);
+            return resolve(true);
+          }
+        }, 200);
+      });
+    }
+
+    const { setIsExecutingCode } = useRunStore.getState();
+    setIsExecutingCode(true);
+    commands.executeCommand("setContext", "SAS.running", true);
+    const session = getSession();
+
+    let logText = "";
+    const onSessionLogFn = session.onSessionLogFn;
+    const onExecutionLogFn = session.onExecutionLogFn;
+    const outputLines = [];
+
+    const addLine = (logLines: LogLine[]) =>
+      outputLines.push(...logLines.map(({ line }) => line));
+
+    try {
+      await session.setup(true);
+
+      // Lets capture output to use it later
+      session.onSessionLogFn = addLine;
```

Comment:
I honestly wasn't super familiar w/ that. I just figured we'd capture all output to prevent it from going to the user. That said, you're right...startup log should be fine to be there. Will adjust.

Reply:
Addressed in https://github.com/sassoftware/vscode-sas-extension/pull/880/commits/7a0e0ad0cbbff2e003227dc7ae40f01c0eb14fc2